pkg/users/password.jsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2020 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+import cockpit from 'cockpit';
+import React from 'react';
+import { FormGroup, Progress, ProgressSize, ProgressMeasureLocation, TextInput } from '@patternfly/react-core';
+
+const _ = cockpit.gettext;
+
+export function password_quality(password, force) {
+    return new Promise((resolve, reject) => {
+        cockpit.spawn('/usr/bin/pwscore', { err: "message" })
+                .input(password)
+                .done(function(content) {
+                    var quality = parseInt(content, 10);
+                    if (quality === 0)
+                        reject(new Error(_("Password is too weak")));
+                    else
+                        resolve(quality);
+                })
+                .fail(function(ex) {
+                    if (!force)
+                        reject(new Error(ex.message || _("Password is not acceptable")));
+                    else
+                        resolve(0);
+                });
+    });
+}
+
+export const PasswordFormFields = ({
+    password, password_confirm,
+    password_label, password_confirm_label,
+    password_strength, password_message,
+    error_password, error_password_confirm,
+    idPrefix, change
+}) => {
+    let variant;
+    if (password_strength === "")
+        variant = "default";
+    else if (password_strength == 100)
+        variant = "success";
+    else if (password_strength > 66)
+        variant = "success";
+    else if (password_strength > 33)
+        variant = "warning";
+    else if (isNaN(password_strength) || password_strength >= 0)
```

Comment:
This condition is a bit weird. I would probably just go with `else`. Or `password_strength < 33) and always resolve with number and never with string.